amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -220,4 +258,53 @@ private void deleteResources(StreamExistenceManager streamExistenceManager, Leas
         log.info("-------------Finished deleting resources.----------------");
     }
 
+    @Data
+    @ToString
+    @AllArgsConstructor
+    public static class StreamScaler implements Runnable {
+        private final KinesisAsyncClient client;
+        private final String streamName;
+        private final Queue<ReshardOptions> scalingFactor;
+        private int totalRotations;
+        private KCLAppConfig testConfig;
+
+        @Override
+        public void run() {
```

Comment:
This is not synchronized, it's a mutable class, and it's scheduled for execution. The only saving grace is the Executor is _currently_ single-threaded, which is not guaranteed to remain that way (e.g., future refactor).